From c9564fafda87d0cbb57208af77dd5c56fab97a69 Mon Sep 17 00:00:00 2001
From: make shi <b15407@freescale.com>
Date: Thu, 8 Nov 2012 15:18:50 +0800
Subject: [PATCH 2246/2463] ENGR00233051-01 Mx6 USB: configure change for OTG
 modulization

- Add USB_FSL_ARC_OTG configuration to imx6_defconfig and imx6s_defconfig,
  the default configuration is selected as "y"
- add related USB_FSL_ARC_OTG configuration to Makefile
- add related USB_FSL_ARC_OTG configuration to Kconfig

Signed-off-by: make shi <b15407@freescale.com>
---
 arch/arm/configs/imx6_defconfig  |    1 +
 arch/arm/configs/imx6s_defconfig |    1 +
 arch/arm/mach-mx6/Kconfig        |    3 +++
 arch/arm/mach-mx6/Makefile       |    3 ++-
 4 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/arch/arm/configs/imx6_defconfig b/arch/arm/configs/imx6_defconfig
index 92cc28b..6f30e36 100644
--- a/arch/arm/configs/imx6_defconfig
+++ b/arch/arm/configs/imx6_defconfig
@@ -316,6 +316,7 @@ CONFIG_MACH_MX6Q_SABREAUTO=y
 #
 # CONFIG_IMX_PCIE is not set
 CONFIG_USB_EHCI_ARC_H1=y
+CONFIG_USB_FSL_ARC_OTG=y
 # CONFIG_MX6_INTER_LDO_BYPASS is not set
 # CONFIG_MX6_CLK_FOR_BOOTUI_TRANS is not set
 CONFIG_ISP1504_MXC=y
diff --git a/arch/arm/configs/imx6s_defconfig b/arch/arm/configs/imx6s_defconfig
index 41f43e5..0c980c9 100644
--- a/arch/arm/configs/imx6s_defconfig
+++ b/arch/arm/configs/imx6s_defconfig
@@ -322,6 +322,7 @@ CONFIG_MACH_MX6Q_SABRESD=y
 #
 # CONFIG_IMX_PCIE is not set
 CONFIG_USB_EHCI_ARC_H1=y
+CONFIG_USB_FSL_ARC_OTG=y
 CONFIG_MX6_INTER_LDO_BYPASS=y
 CONFIG_ISP1504_MXC=y
 # CONFIG_MXC_IRQ_PRIOR is not set
diff --git a/arch/arm/mach-mx6/Kconfig b/arch/arm/mach-mx6/Kconfig
index 97262dc..2ffd90d 100644
--- a/arch/arm/mach-mx6/Kconfig
+++ b/arch/arm/mach-mx6/Kconfig
@@ -261,6 +261,9 @@ config USB_EHCI_ARC_H1
 	tristate "USB Host 1 support"
 	depends on USB_EHCI_ARC
 
+config USB_FSL_ARC_OTG
+	tristate "FSL USB OTG support"
+
 config MX6_INTER_LDO_BYPASS
 	bool "Internal LDO in MX6Q/DL bypass"
 	depends on  REGULATOR_PFUZE100 && CPU_FREQ_IMX && ARCH_MX6
diff --git a/arch/arm/mach-mx6/Makefile b/arch/arm/mach-mx6/Makefile
index 05dc62d..c929f2e 100644
--- a/arch/arm/mach-mx6/Makefile
+++ b/arch/arm/mach-mx6/Makefile
@@ -3,7 +3,7 @@
 #
 
 # Object file lists.
-obj-y   := cpu.o mm.o system.o devices.o dummy_gpio.o irq.o bus_freq.o usb_dr.o usb_h2.o usb_h3.o\
+obj-y   := cpu.o mm.o system.o devices.o dummy_gpio.o irq.o bus_freq.o  usb_h2.o usb_h3.o\
 pm.o cpu_op-mx6.o mx6_wfi.o mx6_fec.o mx6_anatop_regulator.o cpu_regulator-mx6.o \
 mx6_mmdc.o mx6_ddr_freq.o mx6sl_ddr.o mx6sl_wfi.o etm.o
 
@@ -18,3 +18,4 @@ obj-$(CONFIG_SMP) += plat_hotplug.o platsmp.o headsmp.o
 obj-$(CONFIG_LOCAL_TIMERS) += localtimer.o
 obj-$(CONFIG_IMX_PCIE) += pcie.o
 obj-$(CONFIG_USB_EHCI_ARC_H1) += usb_h1.o
+obj-$(CONFIG_USB_FSL_ARC_OTG) += usb_dr.o
\ No newline at end of file
-- 
1.7.7.4

